.github/workflows/selfbuild.yml

```diff
@@ -52,3 +52,49 @@ jobs:
         run: cargo install --path cargo-quickinstall
       - name: Install Thyself with Thyself (or fallback to sensei on windows)
         run: cargo quickinstall cargo-quickinstall || cargo quickinstall sensei
+
+  build-needing-cc:
+    name: Build package that needs CC
+    runs-on: ${{ matrix.os }}
+    defaults:
+      run:
+        shell: bash
+    strategy:
+      matrix:
+        include:
+          # FIXME: cargo-deny doesn't actually build on some arches
+          # - target_arch: x86_64-pc-windows-msvc
+          #   os: windows-latest
+          # - target_arch: x86_64-apple-darwin
+          #   os: macos-latest
+          # - target_arch: aarch64-apple-darwin
+          #   os: macos-latest
+          # - target_arch: x86_64-unknown-linux-gnu
+          #   os: ubuntu-20.04
+          # - target_arch: x86_64-unknown-linux-musl
+          #   os: ubuntu-20.04
+          - target_arch: aarch64-unknown-linux-gnu
+            os: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          persist-credentials: false
+
+      - uses: taiki-e/install-action@v2
+        with:
+          tool: cargo-auditable
+
+      # FIXME: find a package that needs a working CC but doesn't take quite so long to build.
+      - name: Build cargo-deny
```

Comment:
@Milo123459 I am quite tempted to:
* do a full build sweep with the current broken CC
* do a full build sweep with this PR merged in
* get a list of packages that are broken under the broken CC but work under the fixed CC, and put the quickest one in here

Alternatively, we could just not bother merging a regression test for this.

Reply:
I have kicked off a job to attempt builds using what's in main every 5 minutes

`while sleep $((5*60)); do TARGET_ARCHES=aarch64-unknown-linux-gnu ./trigger-package-build.sh ; done`

Quite a lot of packages build just fine, even if CC is not available.

Reply:
hmm, so why is cargo-deny failing? can we make it work?

Reply:
I'm now running the same loop on this branch, to see what builds with this patch applied.

> hmm, so why is cargo-deny failing? can we make it work?

Some packages just fail to build on some architectures. That's just life. We have code to automatically disable packages per-arch if they fail too many times, to avoid wasting CI resources. I've switched it to one that *does* build on most arches and takes about 2 minutes to build. Still worth trying to find a quicker package, but should be good enough to merge.

Reply:
.... hrm... tin-summer's build failure was nothing to do with CC - https://github.com/cargo-bins/cargo-quickinstall/actions/runs/4036562588/jobs/6939242828 . Maybe I should look more closely to find a good package for this CI check.

Reply:
im rerunning it

Reply:
yeah it worked

Reply:
oops: I just cancelled your rerun of https://github.com/cargo-bins/cargo-quickinstall/actions/runs/4036562588 because it looked like it had been running for 2 hours or something

Reply:
on another note: do you know why musl gives so many linker errors? idk of anything we can do really

Reply:
Honestly I don't use musl with rust if I can possibly avoid it (its allocator is unacceptably slow, and debian-slim is good enough for making small docker images these days), so I've not looked into it. Feel free to trawl the logs looking for patterns though. There might be a common theme that's easy to fix.